packages/wallets/neko/src/adapter.ts

```diff
@@ -0,0 +1,215 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+    WalletWindowClosedError,
+} from '@solana/wallet-adapter-base';
+import { Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface NekoWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface NekoWallet extends EventEmitter<NekoWalletEvents> {
+    isNeko?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+}
+
+interface NekoWalletWindow extends Window {
+    neko?: NekoWallet;
+}
+
+declare const window: NekoWalletWindow;
+
+export interface NekoWalletAdapterConfig {}
+
+export const NekoWalletName = 'Neko (Mobile)' as WalletName<'Neko (Mobile)'>;
+
+export class NekoWalletAdapter extends BaseMessageSignerWalletAdapter {
+    name = NekoWalletName;
+    url = 'https://nekowallet.com';
+    icon = 'https://cdn.discordapp.com/attachments/953573404395585596/960393815208914964/Group_7996_1.png';
```

Comment:
This image must be embedded with base64, and should be an SVG. Please heck the other adapters for how this is done.